Alloy/commands/compile/compilerUtils.js

```diff
@@ -665,60 +665,56 @@ exports.copyWidgetResources = function(resources, resourceDir, widgetId, opts) {
 	}
 };
 
-exports.mergeI18n = function(srcI18nDir, compileConfigDir, opts) {
-	logger.info(' i18n:     "' + srcI18nDir + '"');
+exports.mergeI18N = function mergeI18N(src, dest, opts) {
+	var serializer = new XMLSerializer();
+	opts || (opts = {});
 
-	var buildI18nDir = path.join(compileConfigDir.project, CONST.DIR.BUILD_I18N),
-		appI18nDir = path.join(compileConfigDir.project, CONST.DIR.I18N),
-		files = wrench.readdirSyncRecursive(srcI18nDir),
-		serializer = new XMLSerializer();
+	(function walk(src, dest) {
+		if (!fs.existsSync(src)) return;
 
-	if (!fs.existsSync(buildI18nDir)) {
-		wrench.mkdirSyncRecursive(buildI18nDir, 0755);
-		fs.existsSync(appI18nDir) && wrench.copyDirSyncRecursive(appI18nDir, buildI18nDir, {preserve: false});
-	}
+		fs.readdirSync(src).forEach(function (name) {
+			var srcFile = path.join(src, name);
+			var destFile = path.join(dest, name);
 
-	_.each(files, function(file) {
-		var source = path.join(srcI18nDir, file),
-			outputPath = path.join(buildI18nDir, file);
+			if (!fs.existsSync(srcFile)) return;
 
-		if (!path.existsSync(outputPath)) {
-			if (fs.statSync(source).isDirectory()) {
-				wrench.mkdirSyncRecursive(outputPath, 0755);
-			} else {
-				U.copyFileSync(source, outputPath);
+			if (fs.statSync(srcFile).isDirectory()) {
+				fs.existsSync(destFile) || wrench.mkdirSyncRecursive(destFile, 0755);
+				return walk(srcFile, destFile);
 			}
-		} else {
-			if (fs.statSync(source).isFile()) {
-				var outxml = U.XML.parseFromFile(outputPath),
-					root = outxml.documentElement,
-					sourcexml = U.XML.parseFromFile(source),
-					obj = {};
-
-				_.each(outxml.getElementsByTagName('string'), function(node) {
-					var name = node.getAttribute('name');
-					obj[name] = node;
-				});
 
-				_.each(sourcexml.getElementsByTagName('string'), function(node){
-					var name = node.getAttribute('name'),
-						value = node.childNodes[0].nodeValue;
+			if (!fs.existsSync(destFile)) {
```

Comment:
Before this line we first need to skip non-XML files from copying and merging:

```
if (path.extname(name) !== '.xml') return;
```

Or else, if an i18n dir has a `.DS_Store` for example, Alloy will crash with:

```
			_.each(destDoc.getElementsByTagName('string'), function (node) {
			              ^

TypeError: Cannot read property 'getElementsByTagName' of null
    at /Users/fokkezb/dev/forks/alloy/Alloy/commands/compile/compilerUtils.js:700:18
    at Array.forEach (native)
    at walk (/Users/fokkezb/dev/forks/alloy/Alloy/commands/compile/compilerUtils.js:675:23)
    at Object.mergeI18N (/Users/fokkezb/dev/forks/alloy/Alloy/commands/compile/compilerUtils.js:719:3)
    at parseAlloyComponent (/Users/fokkezb/dev/forks/alloy/Alloy/commands/compile/index.js:828:6)
    at /Users/fokkezb/dev/forks/alloy/Alloy/commands/compile/index.js:361:6
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/fokkezb/dev/forks/alloy/Alloy/lib/alloy/underscore.js:79:11)
    at /Users/fokkezb/dev/forks/alloy/Alloy/commands/compile/index.js:350:6
    at Array.forEach (native)
    at Function._.each._.forEach (/Users/fokkezb/dev/forks/alloy/Alloy/lib/alloy/underscore.js:79:11)
    at module.exports (/Users/fokkezb/dev/forks/alloy/Alloy/commands/compile/index.js:346:4)
    at Object.<anonymous> (/Users/fokkezb/dev/forks/alloy/Alloy/alloy.js:113:46)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/Users/fokkezb/dev/forks/alloy/bin/alloy:3:1)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:139:18)
    at node.js:999:3
```